book/src/counter.md

```diff
@@ -0,0 +1,331 @@
+# Building a Counter
+
+This chapter introduces the boxing techniques (pun not intended) for exposing a
+Rust-backed Counter through JNI.
+
+## Define the Java Counter API
+First, we need to define that API that Java will expose. For now, lets say that
+you can fetch the counter and increment it by one. The counter has a few jobs:
+construction, `increment()` , `get()`, and `close()`. Add a stub that looks like
+this:
+
+```java
+class Counter {
+    public Counter() {}
+
+    public int get() {
+    }
+
+    public int increment() {
+    }
+
+    public void close() {
```

Comment:
Shall we clarify here or in the previous paragraph what this method is supposed to do (destroy the native peer)?

Reply:
Yes, let's do that.